### PR TITLE
better bitwise stuff

### PR DESCRIPTION
**Pull request by SansPapyrus683**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as
done._

- [x] I have tested my code
- [x] I have followed the code style guidelines mentioned
      [here](https://usaco.guide/general/adding-solution/#code-conventions)
- [x] I have properly formatted the files according to the steps mentioned
      [here](https://usaco.guide/general/adding-solution#steps)

### CODE REVIEW

File: content/4_Gold/Intro_Bitwise.mdx [hunk 1]
```diff
@@ -49,13 +49,11 @@ both numbers: $a\&b$.
...
-int sum(int a, int b)
-{
-	while(b>0)
-	{
-		int carry=a&b;
-		a^=b;
-		b=carry<<1;
+int add(int a, int b) {
+	while (b > 0) {
+		int carry = a & b;
```
Comment (jessechoe10) [inline content/4_Gold/Intro_Bitwise.mdx:54]: You should write a comment describing what `carry` is

File: content/4_Gold/Intro_Bitwise.mdx [hunk 2]
```diff
@@ -35,8 +35,8 @@ addition/multiplication.
 
 ## Example - Addition
```
Comment (jessechoe10) [inline content/4_Gold/Intro_Bitwise.mdx:36]: Write maybe 1-2 sentences that explains your solution.

**Review by jessechoe10 — CHANGES_REQUESTED** [review]

File: content/4_Gold/Intro_Bitwise.mdx (hunk 1 above)
Comment (SansPapyrus683) [reply content/4_Gold/Intro_Bitwise.mdx:54]: tbh i don't really know either, can someone else explain?

**Comment by dutinmeow** [comment]
Include stuff on `&`, `|`, `~`, `^`, `<<` and `>>` and what they do, instead of just exercises. Also explain concept of storing #s in binary (and how this makes bit operations very fast), and the 2s complement system.

Unrelated but make this a prereq for https://usaco.guide/adv/xor-basis?lang=cpp

**Comment by SansPapyrus683** [comment]
dustin
that isn't my problem
i'm just tryna fix up the code in here

**Comment by jessechoe10** [comment]
> dustin that isn't my problem i'm just tryna fix up the code in here

I'll do it later then but we can merge this lol.

**Review by jessechoe10 — APPROVED** [review]

**Comment by SansPapyrus683** [comment]
@dutinmeow review again

**Review by dutinmeow — APPROVED** [review]

**Review by dongliu0426 — APPROVED** [review]
original was fine but ok

**Pull request merged**